### DISCUSSION TITLE
Pinned favicon spacing issue since 3.9.3

### DISCUSSION THREAD

**Discussion by merchant-ac** [piroor/treestyletab#3213] [Q&A] [answered]
Since version 3.9.3 my carefully crafted layout has been spoilt be ugly gaps between my pinned tabs! The only relevant customisations I have used are:

 /*Change tab height #236, #2389 */

tab-item {
  --tab-size: 18px !important;
  height: var(--tab-size);
   
  :root {
  --favicon-size: 28px;
  --faviconized-tab-size: 18;
}

![2022-09-28 09_22_36-Add-ons Manager — Mozilla Firefox bad](https://user-images.githubusercontent.com/114573373/192744392-c1bea832-2a6e-4d7f-a244-b1d2e59dfe65.jpg)

 Only when I increase --tab-size to 28px or greater do the gaps disappear:

![2022-09-28 09_23_00-Add-ons Manager — Mozilla Firefox good](https://user-images.githubusercontent.com/114573373/192744054-120f25cb-2ad7-4ac5-8bc1-cc8577acc56a.jpg)

This what it looked like in version 3.8.25 (using the older code snippet) before it all went horribly wrong:

![2022-09-28 09_40_32-Add-ons Manager — Mozilla Firefox v3 8 25](https://user-images.githubusercontent.com/114573373/192746415-54ee7b73-e984-4845-9419-ab393e3563dc.jpg)

Please help. Thank you.

Adrian

**Comment by irvinm** [comment, marked as answer]
I didn't dig into the changes near 3.9.3, but I can offer and\or suggest a few things:

First, if you are just looking to decrease the line height, you can modify the CSS to be "non pin tab" specific:

![image](https://user-images.githubusercontent.com/979729/193856482-9530776f-5f38-4ba0-b1bb-22e40792ee67.png)

If you then want to decrease the favico on each of the tabs, just enable the `--favicon-size` selector and set a value.

![image](https://user-images.githubusercontent.com/979729/193856760-ccec2f41-da0f-409c-8bd8-17bfe076b1aa.png)

Finally, as the pinned tabs are basically calculated to be "square" items, you can still change the ratio of the favico to the size of the tab by changing the value `faviconizedTabScale` from the default of 1.75 to something like 1.25 to get smaller pinned tabs.

![image](https://user-images.githubusercontent.com/979729/193857120-8b24ae3e-f5ea-48d1-985b-e8c3f3e4e0ee.png)

**Reply by merchant-ac** [reply]
Thank you! I had completely missed all the extra options in the All Config box at the end of the Options page. Lots to play with there...

My layout has now been restored to its former glory!

Adrian
